src/Translator/Extractor/AdminExtractor.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Translator\Extractor;
+
+use Sonata\AdminBundle\Admin\BreadcrumbsBuilderInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface;
+use Symfony\Component\Translation\Extractor\ExtractorInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * @internal
+ */
+final class AdminExtractor implements ExtractorInterface, LabelTranslatorStrategyInterface
+{
+    private const PUBLIC_ADMIN_METHODS = [
+        'getShow',
+        'getDatagrid',
+        'getList',
+        'getForm',
+    ];
+
+    private const BREADCRUMB_ACTIONS = [
+        'list',
+        'edit',
+        'create',
+        'update',
+        'batch',
+        'delete',
+    ];
+
+    /**
+     * @var string
+     */
+    private $prefix;
```

Comment:
The prefix can be null if `setPrefix` is not called.

Reply:
wouldn't it better to set the default value `''` ? This way it's always a `string`.

setPrefix take only string in the interface ; cf https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Translation/Extractor/ExtractorInterface.php#L36

In next major we'll certainly add typehint `string`.

Reply:
> In next major we'll certainly add typehint `string`.

I don't think we can if we support Symfony 4 & 5